### PR TITLE
Add doctests

### PR DESCRIPTION
**Pull request by mvpatel2000**
Adds doctests with pytest-codeblocks

Future PRs:
- submit PR to run the tests which require calling `trainer.fit()` (will need to add in `marker` checking etc)

### CODE REVIEW

**Comment by mvpatel2000** [comment]
Sequence length warmup examples were commented out in docs but look complete to me... any reason this was the case or is it oversight?

File: STYLE_GUIDE.md [hunk 1]
```diff
@@ -170,6 +185,8 @@ All imports in composer should be absolute -- that is, they do not begin with a
     1.  It must also be specified in the `run_constrained` and the `test.requires` section.
     1.  The import must be conditionally imported in the code. For example:
 
+        <!--pytest-codeblocks:importorskip(monai)-->
+        <!--pytest-codeblocks:importorskip(scikit-learn)-->
         ```python
         from composer.utils import MissingConditionalImportError
```
Comment (mvpatel2000) [inline STYLE_GUIDE.md:191]: Note: This test should pass if the import exists (eg installed with `all`) and raise an error if not installed with UNet support (eg with `dev`). Unfortunately, we can't conditionally change expectation based on packages for doctests, so we just skip the failure case. Same is true for the deepspeed test

**Comment by hanlint** [comment]
@mvpatel2000 @growlix  the SWA README should just be moved from https://github.com/mosaicml/composer/blob/dev/docs/source/method_cards/swa.md as the `README` and a symlink added.

**Review by ravi-mosaicml — COMMENTED** [review]
This is awesome.

Re: the timeouts, I'd still like to be able to run these tests as part of the nightly regression run. Instead of marking the long-running tests as `skip`, thoughts about instead not including the `--codeblocks` as part of the default `make test` command, and instead add a seperate target -- say `make test-codeblocks`? 

Once this is merged, can you update https://github.com/mosaicml/testing/blob/main/composer_nightly/Jenkinsfile to run with the `--codeblocks` flag?

(It'd be awesome if we could somehow have access to marks inside the code block test -- then by default we could mark them all as `@daily`. gpu tests as `@gpu`, etc..., but this is still a great improvement. Perhaps we can contribute to or fork the upstream repo to add this functionality in?)

**Comment by mvpatel2000** [comment]
> @mvpatel2000 @growlix the SWA README should just be moved from https://github.com/mosaicml/composer/blob/dev/docs/source/method_cards/swa.md as the `README` and a symlink added.

Matthew said there are a few parameter updates missing from the readme?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -18,13 +18,15 @@ To set up the development environment in your local box, run the commands below.
 
 1\. Install the dependencies needed for testing and linting the code:
 
+<!--pytest-codeblocks:skip-->
```
[the remark this answers is not in the document]
Comment (mvpatel2000) [reply CONTRIBUTING.md:21]: It also tests bash

[the remark this answers is not in the document]
Comment (ravi-mosaicml) [reply CONTRIBUTING.md:21]: Ohh, like it runs bash scripts too? Neat

**Comment by mvpatel2000** [comment]
> This is awesome.
> 
> Re: the timeouts, I'd still like to be able to run these tests as part of the nightly regression run. Instead of marking the long-running tests as `skip`, thoughts about instead not including the `--codeblocks` as part of the default `make test` command, and instead add a seperate target -- say `make test-codeblocks`?
> 
> Once this is merged, can you update https://github.com/mosaicml/testing/blob/main/composer_nightly/Jenkinsfile to run with the `--codeblocks` flag?
> 
> (It'd be awesome if we could somehow have access to marks inside the code block test -- then by default we could mark them all as `@daily`. gpu tests as `@gpu`, etc..., but this is still a great improvement. Perhaps we can contribute to or fork the upstream repo to add this functionality in?)

Discussed offline, there's a few ideas around using `skipif` to run these codeblocks. It will take some time though, so pro-pushing to a second PR after 0.7 release

**Review by ravi-mosaicml — APPROVED** [review]
LGTM; looking forward to removing those additional skips

**Pull request merged**